### PR TITLE
[WIP] Refactor internal functionality

### PR DESCRIPTION
**Pull request by davidlumley**
Currently in progress.

The intent of this is to leave the public API relatively unchanged, but tidy up the internal functionality to make it simpler to understand, and as a result make it simpler to test and to modify.

The most interesting change is probably how the model classes are designed, and how the client interfaces with them through the `Query` class. This allows method chaining (like `client.model.where(condition).limit(10).all`), and also helps separate functionality.

### CODE REVIEW

File: lib/myob/api/client.rb [hunk 1]
```diff
@@ -1,69 +1,43 @@
...
+            Myob::Api::Query.new(model_class)
+          end
         end
-      end
 
-      def get_access_code_url(params = {})
-        @client.auth_code.authorize_url(params.merge(scope: 'CompanyFile', redirect_uri: @redirect_uri))
+        def models
+          @models ||= [
+            Myob::Api::Model::Contact,
```
Comment (ghiculescu) [inline lib/myob/api/client.rb]: I noticed you removed a whole bunch of models, should they go here?

Comment (davidlumley) [inline lib/myob/api/client.rb]: Yup.

Not sure if it's better to make it explicit about which objects are loaded, or should make use of ObjectSpace like @BrunoChauvet does here https://github.com/davidlumley/myob-api/pull/19/files#diff-0107adb2ba05918844bfaaba5ec45f18R12

Comment (ghiculescu) [inline lib/myob/api/client.rb]: I like the `ObjectSpace

File: myob-api.gemspec [hunk 2]
```diff
@@ -21,5 +21,6 @@ Gem::Specification.new do |spec|
   spec.add_dependency "oauth2", "> 0.8"
 
   spec.add_development_dependency "bundler", "~> 1.3"
+  spec.add_development_dependency "minitest", "~> 5.4"
```
Comment (davidlumley) [inline myob-api.gemspec]: There's an issue with CircleCI regarding minitest versions and (at the minimum) ruby 2.2. Being explicit about which version should be used clears this up.

**Pull request closed without merging**